packages/codegen-ui/lib/generate-form-definition/form-to-component.ts

```diff
@@ -88,7 +88,30 @@ export const fieldComponentMapper = (name: string, formDefinition: FormDefinitio
   return parentGrid(`${name}Grid`, formDefinition.form.layoutStyle, fieldChildren);
 };
 
-export const ctaButtonConfig = (): StudioComponentChild => {
+const resolveCtaLabels = (
+  formDefinition: FormDefinition,
+): { cancelLabel: string; clearLabel: string; submitLabel: string } => {
+  const cancelLabel =
+    formDefinition.buttons?.cancel?.visible && formDefinition.buttons?.cancel?.labelOverride
+      ? formDefinition.buttons?.cancel?.labelOverride
+      : 'Cancel';
+
+  const clearLabel =
```

Comment:
Please move the defaults to the default object in the form definition mapper. `helpers/defaults.ts`

Reply:
This way, the UI will get the defaults too and we won't have a discrepancy. 
General note - you will probably want to do some mapping in the definition mapper (to apply the defaults, and also to figure out the position of the buttons in relation to one another)

Reply:
moved defaults to _helpers/defaults.ts_

Reply:
That is definitely one part. What remains is applying the defaults within the definition mapper. You can look at how inputs use the defaults, for example. You will want to create a helper that--among other things--applies these defaults so buttons always have labels, both in the UI and in the customer project. 

It would be best if the shape of the buttons config produced by the definition mapper is different from the buttons config passed in... a shape that is more digestible by the form-to-components helpers. Again, you can see how inputs are mapped for example

Reply:
Thanks for the explanation, it's clearer to me now how defaults work with definition mapper.  I've added _mapButtons_ to a new helper file to set defaults